File: frontend/lib/features/news/data/news_item.dart
class NewsItem {
  const NewsItem({
    required this.id,
    required this.title,
    required this.link,
    required this.sourceName,
    this.excerpt,
    this.publishedAt,
  });

  final String id;
  final String title;
  final String link;
  final String sourceName;
  final String? excerpt;
  final DateTime? publishedAt;

  factory NewsItem.fromJson(Map<String, dynamic> json) => NewsItem(
        id: json['id'] as String,
        title: json['title'] as String,
        link: json['link'] as String,
        sourceName: json['source_name'] as String,
        excerpt: json['excerpt'] as String?,
        publishedAt: json['published_at'] == null ? null : DateTime.parse(json['published_at'] as String),
      );
}

File: backend/app/models/news_item.py
"""NewsItem — one persisted, source-attributed feed entry, see
docs/08-modules/news-intelligence.md.

`title` and `excerpt` are always the source's own RSS text, verbatim —
never AI-generated (see docs/adr/0014-defer-ai-summarization.md: measured
and deferred, not a design preference). `link` is unique — the idempotency
key NewsService upserts on (see docs/08-modules/news-intelligence.md
"Idempotence, not Deduplication" for why this isn't the deferred
semantic-deduplication pipeline stage).
"""
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, String, Text, Uuid
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class NewsItem(Base):
    __tablename__ = "news_items"

    id: Mapped[uuid.UUID] = mapped_column(Uuid, primary_key=True, default=uuid.uuid4)
    source_id: Mapped[uuid.UUID] = mapped_column(
        Uuid, ForeignKey("sources.id", ondelete="CASCADE"), nullable=False, index=True
    )

    title: Mapped[str] = mapped_column(String(500), nullable=False)
    link: Mapped[str] = mapped_column(String(2000), nullable=False, unique=True, index=True)
    excerpt: Mapped[str | None] = mapped_column(Text, nullable=True)
    published_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    fetched_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), nullable=False
    )
